### ISSUE TITLE
Final tasks for Q1

### ISSUE THREAD

**Issue by rcurrie** [anthem-ai/ucsc-engineering-2020#3]
Following up on our weekly call with notes on the tasks and basic recipes

**Ingest** Develop a pipeline that can go through sample (fhir.json.gz) one by one and export back to disk a set of features per sample. For example the time bucketed one-hot for that sample, or after you have embeddings the sequence of embeddings for that sample. One common pattern would be to store back down to disk, in another folder, .npy (binary numpy) files per sample after the above transformations so that they are all ready for the next group. You may want to encode in a separate .npy, or just in the name of the resulting sample, the label... You can use PyTorch DataSet and DataLoader, or just a for loop...

**Train** Now that you have, for example, two folders of .npy files - one with one-hot and another with embeddings, run a CNN vs. RNN training over them. You should use DataSet and DataLoader to run over the samples to feed training. Experiment with stratified train/test split to get a better more regularized model.

**Visualize** Select a small (~1000) sample of patients randomly, average their embeddings into a single vector per Patient and plot with t-SNE.

**Comment by rcurrie** [comment]
[Flagging suspicious healthcare claims with Amazon SageMaker](https://aws.amazon.com/blogs/machine-learning/flagging-suspicious-healthcare-claims-with-amazon-sagemaker/) article with a notebook that computes embeddings from with gensim as well as plots via t-SNE. There are also other projects out there specifically focused on codes to vectors - if you find a pre-computed set of embeddings feel free to use that but its also worthwhile to just run gensim yourself.

**Comment by rcurrie** [comment]
Here's a quick notebook (extracted from a larger one) that illustrates using the python fhir library and pytorch datasets to batch through and extract codes into text files (leaving converting to one-hot and embedding numpy to the reader :-).
[fhir-to-code-npy.ipynb.zip](https://github.com/anthem-ai/ucsc-engineering-2020/files/4300249/fhir-to-code-npy.ipynb.zip)
